### ISSUE TITLE
DefaultParameterArgument value can't be suppressed

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#47348]
**Version Used**: e7bad77

**Steps to Reproduce**:

```cs
#nullable enable
using System.Runtime.InteropServices;

public class C {
    public void M1([Optional, DefaultParameterValue(null!)] object obj) // error
    {
        obj.ToString();
    }
}
```

**Expected Behavior**: No warnings

**Actual Behavior**: error CS8017: The parameter has multiple distinct default values.

Related to #43399
